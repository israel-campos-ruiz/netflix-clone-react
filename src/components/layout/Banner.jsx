import React, { useEffect, useState } from "react";
import axios from "axios";
import { requests, URL_POSTER } from "../../helpers/request";
import '../../styles/components/layout/Banner.css'
const Banner = () => {
  const [movie, setMovie] = useState([]);
  useEffect(() => {
    const fetchData = async () => {
      const request = await axios.get(requests.fetchNetflixOriginals);
      setInterval(() => {
        setMovie( request.data.results[Math.floor(Math.random() * request.data.results.length - 1 )]                            
        );
      }, 8000);      
    };
    fetchData();
    return () => {
       
    }
   }, []);
   function truncate(str,n){
       return str?.length > n ? str.substr(0, n-1) + "..." : str;
   }
  return (
    <header
      className="banner"
      style={{
        backgroundSize: "cover",
        backgroundImage: `url(${URL_POSTER}${movie?.backdrop_path})`,
        backgroundPosition: "center center",
        transitionTimingFunction: "ease-in-out",
        transition: "all 0.8s",
      }}
    >
      <div className="banner__contents">
        <h1 className="banner__title">{movie?.title || movie?.name || movie?.original_name}</h1>
        <div className ="banner__buttons">
            <button className="banner__button">Play</button>
            <button className="banner__button">My list</button>
        </div>
        {/* description */}
        <h1 className="banner__description">
            {truncate(movie?.overview, 200)}
        </h1>


      </div>
      <div className="banner--fadeBottom"></div>
    </header>
  );
};

export default Banner;
