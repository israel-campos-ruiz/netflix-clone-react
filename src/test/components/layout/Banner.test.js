import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import Banner from '../../../components/layout/Banner';

function truncate(str,n){
    return str?.length > n ? str.substr(0, n-1) + "..." : str;
}
describe('Pruebas en el componente <Banner />', () => {
    const wrapper = shallow(<Banner />);
    test('debe hacer crear el snapshot', () => {
        expect(wrapper).toMatchSnapshot()
    });

   test('Debe de cortar el parrafo con la cantidad de caracteres que le pasemos', () => {
      expect(truncate('palabra',6)).toBe('palab...')
   });
   
});