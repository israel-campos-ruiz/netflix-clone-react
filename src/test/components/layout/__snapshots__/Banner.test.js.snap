// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <Banner /> debe hacer crear el snapshot 1`] = `
<header
  className="banner"
  style={
    Object {
      "backgroundImage": "url(https://image.tmdb.org/t/p/original/undefined)",
      "backgroundPosition": "center center",
      "backgroundSize": "cover",
      "transition": "all 0.8s",
      "transitionTimingFunction": "ease-in-out",
    }
  }
>
  <div
    className="banner__contents"
  >
    <h1
      className="banner__title"
    />
    <div
      className="banner__buttons"
    >
      <button
        className="banner__button"
      >
        Play
      </button>
      <button
        className="banner__button"
      >
        My list
      </button>
    </div>
    <h1
      className="banner__description"
    />
  </div>
  <div
    className="banner--fadeBottom"
  />
</header>
`;
